main.go

```diff
@@ -18,45 +22,130 @@ var posts = []post{{
 	Title: "YOusdkfkd",
 }}
 
-func postsHandler(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-
-	handlers := map[string]http.HandlerFunc{
-		"GET": func(w http.ResponseWriter, r *http.Request) {
-			resp, _ := json.Marshal(posts)
-			fmt.Fprintf(w, string(resp))
-		},
-		"POST": func(w http.ResponseWriter, r *http.Request) {
-			decoder := json.NewDecoder(r.Body)
-			var p post
-			if err := decoder.Decode(&p); err != nil {
-				panic(err)
-			}
-
-			id := len(posts) + 1
-			p.ID = id
-			posts = append(posts, p)
-			resp, _ := json.Marshal(p)
-			fmt.Fprintf(w, string(resp))
-		},
-		"DELETE": func(w http.ResponseWriter, r *http.Request) {
-			println(r.URL.Path)
-		},
-	}
-
-	handler, ok := handlers[r.Method]
-	if !ok {
-		fmt.Fprintf(w, "Method not supported")
+func find(id int) (int, error) {
+	for i, v := range posts {
+		if v.ID == id {
+			return i, nil
+		}
+	}
+
+	return 0, errors.New("Not found")
+}
+
+func getPosts(w http.ResponseWriter, r *http.Request) {
+	resp, err := json.Marshal(posts)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Fprintf(w, string(resp))
```

Comment:
`resp` is `[]byte`, so you can just do
```suggestion
	_, err := w.Write(resp)
	if err != nil {
		http.Error(w, "failed to write response", http.StatusInternalServerError)
	}
```